rotkehlchen/externalapis/etherscan_like.py

```diff
@@ -0,0 +1,328 @@
+    def _query(
+            self,
+            chain_id: SUPPORTED_CHAIN_IDS,
+            module: str,
+            action: str,
+            options: dict[str, Any] | None = None,
+            timeout: tuple[int, int] | None = None,
+    ) -> list[dict[str, Any]] | str | int | dict[str, Any] | None:
+        ...
+
+    @abstractmethod
+    def get_blocknumber_by_time(
+            self,
+            chain_id: SUPPORTED_CHAIN_IDS,
+            ts: Timestamp,
+            closest: Literal['before', 'after'] = 'before',
+    ) -> int:
+        ...
+
+    def _process_timestamp_or_blockrange(
+            self,
+            chain_id: SUPPORTED_CHAIN_IDS,
+            period: TimestampOrBlockRange,
+            options: dict[str, Any],
+    ) -> dict[str, Any]:
+        """Process TimestampOrBlockRange and populate call options"""
+        if period.range_type == 'blocks':
+            from_block = period.from_value
+            to_block = period.to_value
+        else:  # timestamps
+            from_block = self.get_blocknumber_by_time(
+                chain_id=chain_id,
+                ts=period.from_value,  # type: ignore
+                closest='before',
+            )
+            to_block = self.get_blocknumber_by_time(
+                chain_id=chain_id,
+                ts=period.to_value,  # type: ignore
+                closest='before',
+            )
+
+        options['startblock'] = str(from_block)
+        options['endblock'] = str(to_block)
+        return options
+
+    @staticmethod
+    def _maybe_paginate(
+            result: list[dict[str, Any]],
+            options: dict[str, Any],
+    ) -> dict[str, Any] | None:
+        """Check if the results have hit the pagination limit.
+        If yes adjust the options accordingly. Otherwise signal we are done"""
+        if len(result) != ETHERSCAN_QUERY_LIMIT:
+            return None
+
+        # else we hit the limit. Query once more with startblock being the last
+        # block we got. There may be duplicate entries if there are more than one
+        # entries for that last block but they should be filtered
+        # out when we input all of these in the DB
+        last_block = result[-1]['blockNumber']
+        options['startblock'] = last_block
+        return options
+
+    @overload
+    def get_transactions(
+            self,
+            chain_id: SUPPORTED_CHAIN_IDS,
+            account: ChecksumEvmAddress | None,
+            action: Literal['txlistinternal'],
+            period_or_hash: TimestampOrBlockRange | EVMTxHash | None = None,
+    ) -> Iterator[list[EvmInternalTransaction]]:
+        ...
+
+    @overload
+    def get_transactions(
+            self,
+            chain_id: SUPPORTED_CHAIN_IDS,
+            account: ChecksumEvmAddress | None,
+            action: Literal['txlist'],
+            period_or_hash: TimestampOrBlockRange | EVMTxHash | None = None,
+    ) -> Iterator[list[EvmTransaction]]:
+        ...
+
+    def get_transactions(
```

Comment:
All these functions in this file have not changed from before in implementation? How much should I check them?